### PR TITLE
refactor(shred-networking): move turbine into shred_collector 

### PR DESCRIPTION
**Pull request by yewman**
Consolidates logic associated with the transmission of shreds in Solana by  moving the implementation of turbine tree and retransmit stage into the shred collector module, which is renamed to **shred_networking** 

Changes in this PR are kept **minimal**, refactoring of data-structures and dependencies can be addressed in follow up PR's.

### CODE REVIEW

**Comment by dnut** [comment]
shred_network was my first thought on a name for this but it doesn't feel totally satisfying because networking is only a part of what this code is doing. It also has some basic logic for validating shreds, routing them internally to other components such as the blockstore, and determining which peers should receive the shreds.

Do any of these names seem like an improvement? I'm leaning towards shred_router or shred_dispatcher but I think any of these could work.

- shred_router
- shred_dispatcher
- shred_coordinator
- shred_pipeline
- shred_plumbing
- shred_io
- shred_director
- shred_navigator
- shred_orchestrator
- shred_hub
- shred_relay
- shred_switchboard

**Comment by 0xNineteen** [comment]
imo shred_networking is still the best - the core logic is recv/send from the network which networking describes, extra logic is fine

**Review by 0xNineteen — DISMISSED** [review]

**Review by 0xNineteen — APPROVED** [review]

File: src/cmd/cmd.zig [hunk 1]
```diff
@@ -821,27 +821,10 @@ fn validator() !void {
...
-        .gossip_table_rw = &gossip_service.gossip_table_rw,
-        .receiver = &retransmit_shred_channel,
-        .maybe_num_retransmit_threads = config.current.turbine.num_retransmit_threads,
-        .overwrite_stake_for_testing = config.current.turbine.overwrite_stake_for_testing,
-        .exit = &app_base.exit,
-        .rand = prng.random(),
-        .logger = app_base.logger.unscoped(),
-    }});
-
     // shred collector
```
Comment (dadepo) [inline src/cmd/cmd.zig:824]: Nit: comment was missed.

File: src/cmd/cmd.zig [hunk 2]
```diff
@@ -951,28 +937,10 @@ fn shredCollector() !void {
...
```
Comment (dadepo) [inline src/cmd/cmd.zig:940]: Nit: comment was missed.

File: src/ledger/readme.md [hunk 3]
```diff
@@ -57,7 +57,7 @@ For more information on Shreds, see the spec [here](https://github.com/solana-fo
 
 The ShredCollector is responsible for gathering and storing shreds from the network. While it is not a direct
```
Comment (dadepo) [inline src/ledger/readme.md:58]: Nit: Should this section of the documentation be renamed to emphasis Shred Network instead of ShredCollector? Or maybe rephrased to emphasis the task of collecting shreds from the network by the shred network module?

File: src/ledger/readme.md [hunk 4]
```diff
@@ -72,7 +72,7 @@ graph TD
     D --> E[BlockstoreReader]
 ```
 
-![ShredCollector Component](./imgs/shred_collector_component.png)
+![ShredCollector Component](./imgs/shred_network_component.png)
```
Comment (dadepo) [inline src/ledger/readme.md:75]: I think the actual renaming of the file was missed. So this change probably has lead to a broken link
